components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/validators/AbstractResponseTypeRequestValidator.java

```diff
@@ -0,0 +1,274 @@
+                            OAuthConstants.LogConstants.FAILED, "OAuth application is not in active state.",
+                            "validate-input-parameters", null);
+                }
+                throw new InvalidOAuthClientException("Oauth application is not in active state");
+            }
+            return validateCallBack(clientId, callbackURI, appDO);
+        } catch (InvalidOAuthClientException e) {
+            // There is no such Client ID being registered. So it is a request from an invalid client.
+            if (log.isDebugEnabled()) {
+                log.debug("Error while retrieving the Application Information", e);
+            }
+            if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                Map<String, Object> params = new HashMap<>();
+                params.put("clientId", clientId);
+                LoggerUtils.triggerDiagnosticLogEvent(OAuthConstants.LogConstants.OAUTH_INBOUND_SERVICE, params,
+                        OAuthConstants.LogConstants.FAILED,
+                        "Cannot find an application associated with the given client_id",
+                        "validate-oauth-client", null);
+            }
+            OAuth2ClientValidationResponseDTO validationResponseDTO = new OAuth2ClientValidationResponseDTO();
+            validationResponseDTO.setValidClient(false);
+            validationResponseDTO.setErrorCode(OAuth2ErrorCodes.INVALID_CLIENT);
+            validationResponseDTO.setErrorMsg(e.getMessage());
+            return validationResponseDTO;
+        } catch (IdentityOAuth2Exception e) {
+            log.error("Error when reading the Application Information.", e);
+            LoggerUtils.triggerDiagnosticLogEvent(OAuthConstants.LogConstants.OAUTH_INBOUND_SERVICE, null,
+                    OAuthConstants.LogConstants.FAILED, "Server error occurred.",
+                    "validate-input-parameters", null);
+            OAuth2ClientValidationResponseDTO validationResponseDTO = new OAuth2ClientValidationResponseDTO();
+            validationResponseDTO.setValidClient(false);
+            validationResponseDTO.setErrorCode(OAuth2ErrorCodes.SERVER_ERROR);
+            validationResponseDTO.setErrorMsg("Error when processing the authorization request.");
+            return validationResponseDTO;
+        }
+    }
+
+    private OAuth2ClientValidationResponseDTO validateCallBack(String clientId, String callbackURI, OAuthAppDO appDO) {
+
+        if (!parametersToValidate.contains(REDIRECT_URI)) {
+            OAuth2ClientValidationResponseDTO validationResponseDTO = new OAuth2ClientValidationResponseDTO();
+            validationResponseDTO.setValidClient(true);
+            validationResponseDTO.setApplicationName(appDO.getApplicationName());
+            validationResponseDTO.setPkceMandatory(appDO.isPkceMandatory());
+            validationResponseDTO.setPkceSupportPlain(appDO.isPkceSupportPlain());
```

Comment:
why are we setting these inside a callback url validation method?

Reply:
There are two more places below thats setting these. Seems to not belong to this method. As I can see this code is from a previous implementation, but lets check and see why we need them here. It doesn't seem correct to set unrelated params during a callback validation